source/fab/steps/grab.py

```diff
@@ -157,3 +155,40 @@ def run(self, artefact_store: Dict, config):
 
             r = remote.RemoteClient(self.src)
             r.export(str(config.source_root / self.dst_label), revision=self.revision, force=True)
+
+
+class GrabPreBuild(Step):
+    """
+    Copy the contents of another pre-build folder into our own.
+
+    """
+    def __init__(self, path, objects=True, allow_fail=False):
+        super().__init__(name=f'GrabPreBuild {path}')
+        self.src = path
+        self.objects = objects
+        self.allow_fail = allow_fail
+
+    def run(self, artefact_store: Dict, config):
+        dst = config.prebuild_folder
+        try:
+            res = call_rsync(src=self.src, dst=dst)
+
+            # log the number of files transferred
+            to_print = [line for line in res.decode().split('\n') if 'Number of' in line]
+            logger.info('\n'.join(to_print))
+
+        except RuntimeError as err:
+            msg = f"could not grab pre-build '{self.src}': {err}"
+            if not self.allow_fail:
+                raise RuntimeError(msg)
+            logger.warning(msg)
+
+
+def call_rsync(src: Union[str, Path], dst: Union[str, Path]):
+    # we want the source folder to end with a / for rsync because we don't want it to create a sub folder
+    src = os.path.expanduser(str(src))
+    if not src.endswith('/'):
+        src += '/'
+
+    command = ['rsync', '-times', '--stats', '-ru', src, str(dst)]
```

Comment:
`--times`?

```suggestion
    command = ['rsync', '--times', '--stats', '-ru', src, str(dst)]
```

Reply:
Its `-t` or `--times`.
https://linux.die.net/man/1/rsync

Reply:
Wait it _wasn't_ already `--`?! Ok thanks!
That makes me think it's no longer needed for the test,
but still feels like the right thing to do, so leaving in.